include/powsybl/iidm/Network.hpp

```diff
@@ -150,6 +150,19 @@ class Network : public Container, public VariantManagerHolder {
 
     const stdcxx::DateTime& getCaseDate() const;
 
+    template <typename T, typename = typename std::enable_if<std::is_base_of<Connectable, T>::value>::type>
+    unsigned long getConnectableCount() const;
```

Comment:
I think you should also add a non template version, or add a default value to T. This will allow the use to write `network.getConnectableCount();`
I think with your proposal, it's not possible, a user has to write `network.getConectableCount<Connectable>()`